### DISCUSSION TITLE
Exporting to osm.pbf

### DISCUSSION THREAD

**Discussion by luukvdmeer** [ropensci/osmextract#268] [Q&A]
Hi! With `osmextract` you can import OSM data from an osm.pbf file. Then, you can make changes to the data. E.g. reversing geometries, updating attributes, etc. My question is: after making this changes, is it possible to export the wrangled data again to the original (or a new one) osm.pbf file? Have you experimented with such a workflow? cc @Robinlovelace 

Or to put it differently: Can you use `osmextract` to edit an existing osm.pbf file?

**Comment by Robinlovelace** [comment]
This would be useful but, if I recall correctly, GDAL (and therefore sf) cannot write to .osm.pbf and it would need an intermediary tool. Happy to be told that is not right though!

**Comment by agila5** [comment]
I've never seen an example where someone is writing or editing existing `.osm.pbf` files (probably for the same reasons that Robin described). It might be really useful to implement an `oe_write()` function following the guidelines described [here](https://wiki.openstreetmap.org/wiki/PBF_Format), but I'm not going to work on `osmextract` for the next weeks/months, sorry 😢
